shard/src/ext/pug_cog.py

```diff
@@ -0,0 +1,59 @@
+from discord.ext import commands
+from contextlib import suppress
+import time
+import asyncio
+from lib.config import logger
+
+
+class Pug(commands.Cog):
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command()
+    async def startpug(self, ctx, requiredPlayers: int = 10):
+        '''
+        Starts a tally for pugs
+        '''
+        settings = self.bot.settings[ctx.guild]
+        pug_emoji = settings.pug_emoji
+
+        t_end = time.time() + 60 * 30
+        user_list = []
+        msg = await ctx.send(f"{requiredPlayers} more {pug_emoji}'s for pugs")
+        await msg.add_reaction(pug_emoji)
+        while time.time() < t_end:
+
+            def check(r, u):
+                return r.message.id == msg.id and str(r.emoji) == pug_emoji
+
+            with suppress(TimeoutError):
+                task_add = asyncio.ensure_future(self.bot.wait_for('reaction_add', check=check))
+                task_remove = asyncio.ensure_future(self.bot.wait_for('reaction_remove', check=check))
+                done, _ = await asyncio.wait([task_add, task_remove], timeout=5, return_when=asyncio.FIRST_COMPLETED)
+
+                if task_add in done:
+                    task_remove.cancel()
+                    react, user = task_add.result()
+                    t_end += int((settings.pug_timeout_speed / 2) * 60)
+                elif task_remove in done:
+                    task_add.cancel()
+                    react, user = task_remove.result()
+                else:
+                    continue
+
+            user_list = [u for u in await react.users().flatten() if u != self.bot.user]
+            num_left = max(0, (requiredPlayers - len(user_list)))
+            await msg.edit(content=f"{num_left} more {pug_emoji}'s for pugs")
```

Comment:
I could add a time remaining thingy here for clarity but there is a bug right now that allows someone to make the timer go up by constantly reacting and unreacting so it might not be best to show seconds remaining.